sdk/src/math/auction.ts

```diff
@@ -104,3 +104,21 @@ export function getAuctionPriceForOracleOffsetAuction(
 
 	return oraclePrice.add(priceOffset);
 }
+
+export function deriveOracleAuctionParams({
+	oraclePrice,
+	auctionStartPrice,
+	auctionEndPrice,
+	limitPrice,
+}: {
+	oraclePrice: BN;
+	auctionStartPrice: BN;
+	auctionEndPrice: BN;
+	limitPrice: BN;
+}): { auctionStartPrice: BN; auctionEndPrice: BN; limitPrice: BN } {
+	return {
+		auctionStartPrice: auctionStartPrice.sub(oraclePrice),
+		auctionEndPrice: auctionEndPrice.sub(oraclePrice),
+		limitPrice: limitPrice.sub(oraclePrice),
+	};
```

Comment:
might want to consider making them non-zero (if they are zero?) like -1 or +1 if its zero for start/end/limit?